src/goto-symex/symex_complexity.cpp

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+#include "symex_complexity.h"
+#include "goto_symex_state.h"
+
+bool symex_complexityt::is_too_complex(goto_symex_statet &state)
```

Comment:
Although `is_too_complex` is a good name for a >100 line function ;-), I guess it doesn't properly convey very well that its main effect is a side-effect.

Reply:
Agreed, perhaps name this `check_complexity`